### DISCUSSION TITLE
是时候结束游戏了。/ It's time to end the game.

### DISCUSSION THREAD

**Discussion by mywalkb** [LSPosed/LSPosed#1657] [General]
I copied the title from [vvb2060](https://github.com/vvb2060/MagiskDetector)
I'm following this project since the born after the fork from EdXposed because the mainteiner didn't accept your PR.
Every day new features, fast to release and fast to resolve issues. Very great project!
Something has happened lately, something bad.
The feature request #9 (Module API) and #405 (CLI) have been boycotted.
The PR #1593 which introduce CLI, before is set as draft and after the owner tired to wait has been closed it. Why don't accept it? Cli was working. He develope also the suggestions of @yujincheng08 
Why don't you want an external app to access the manager? I can understand the security, but the device is mine, the installation is mine, let me to choose for my device, I'm not a dumb user. I don't use untrusted module, only trusted. I developed a module is not public, because the ui is bad, the scope is dinamic, in my ui I choose the app to hook, in the past everytime I forgot to set the same app in manager for my module and I solved with direct access to db manager. Why can't i do it from my module? It's mine is trusted.
I think every user should choose the best configuration for himself. Dictatorship is bad.
The last version improve the ui, but was it necessary? I think was more important module api and cli. I don't want choose for the developers. I used the ui, about 5 minutes in a year, while the module api is more useful.
If you continue on this path you behave exactly like the mainteiner of edxposed.
Another fork is it useful?

**Comment by yujincheng08** [comment]
> Why don't accept it

It's not we that close it.

> Why don't you want an external app to access the manager?

Access the manager to do what? Detect it?

> I'm not a dumb user

You are if you don't submit a PR for what you want but order us like you are the employer.
And not every user is not dumb. It is not a project only for you.

> Dictatorship is bad

It's an open-source project tho.

> I think was more important module API

Yes. But designing an API is not easy.


BTW, your English is really bad. Consider using Grammarly or so to improve it so that we can understand what on earth you are saying.

![image](https://user-images.githubusercontent.com/5022927/153251333-6fc0ed1b-e964-4422-826c-7866fd094382.png)
